src/Sarif/ExtensionMethods.cs

```diff
@@ -185,7 +185,19 @@ public static string GetMessageText(this Result result, IRule rule, bool concise
             {
                 throw new ArgumentNullException(nameof(rule));
             }
+            return GetMessageText(result, rule, concise: false);
```

Comment:
A subtlety: When you have an overload that is just a _pass-through_ to an overload with more arguments, do all the argument checking in the overload with the most arguments. That way, you only have to write the argument-checking code once. So remove this check... unless... do you get another warning if you remove this check?

Also, formatting nitpick: our convention is to add a blank line after a closing brace. Likewise after line 200.